platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,826 @@
+`tssEcKey` that is used in the Groth21 algorithm.
+
+TODO: Protobuf of data
+
+##### Key Material (TSS Data Map)
+
+The TSS Data Map is a Map Merkle Leaf that is a combined map of all the key material for all rosters that are
+tracked and the votes by nodes that have validated the key material.
+
+Keys in the TSS Data Map have the following structure: (KeyType, RosterHash, SequenceNumber).
+
+1. KeyType - An enum that is one of TSS_MESSAGE or TSS_VOTE.
+2. RosterHash - The hash of the consensus roster that the message or vote is related to.
+3. SequenceNumber - The order in which the message or vote came to consensus.
+
+The value associated with a TSS_MESSAGE key is the pair (ShareId, TssMessage)
+
+1. ShareId - The id of the share that the message is related to.
+2. TssMessage - the raw bytes of the TssMessage
+
+The value associated with a TSS_VOTE key is a bit vector with the following interpretation
+
+* The order of bits from the least significant bit to the most significant bit corresponds to the numeric order of
+  share ids from least to greatest.
+* If a bit is set to 1, then the TssMessage for the corresponding ShareId was valid and contributed to a successful
+  reconstruction of the ledger id.
+* If a bit is set to 0, then the TssMessage for the corresponding ShareId was either invalid, was not received, or
+  was not used in the reconstruction of the ledger id.
+* If the sum of the bits is greater than the threshold, then the vote is a yes vote in favor of adopting the
+  candidate roster.
+
+Lifecycle Invariants
+
+1. Inserts should only be made for candidate rosters that are going through the keying process.
+2. When a roster is no longer stored in the state, the corresponding key material should be removed.
+3. Every roster that has become active after the TSS bootstrap process must have sufficient key material to be
+   able to recover the ledger id.
+
+TODO: Protobuf of data
+
+##### Partial Ledger Signatures
+
+For a configured number of rounds, share signatures on a message are stored in the state until a threshold number of
+them have been collected to produce the ledger id. Each time a signature on a message is added to the collection,
+the round counter resets. Once the ledger id is produced, it is stored with the other signatures and no new
+signatures are added to the collection. Once a configured number of rounds have passed after the last signature
+added to a collection for a message, the signatures are removed from the state.
+
+The partial ledger signatures are stored in a Map Merkle Leaf where the keys are the message hash and the values are
+a tuple (List<ShareSignature>, LedgerSignature, lastRoundUpdated).
+
+TODO: Protobuf of data
+
+### Core Behaviors
+
+The new behavior is related to the bootstrap process of creating the ledger id, transferring the ability to sign
+messages from one roster to the next roster, and creating ledger signatures.
+
+#### TSS Bootstrap for New Networks
```

Comment:
It seems a bit odd that there are 2 different sections in this document describing bootstrapping. There is this of the heading under `Core Behaviors`, but there's a different section under `Design Decisions`. I'm not clear on what the difference is, and why which information goes in which section.

Reply:
I have modified this section to focus on the startup sequence in detail.